tmt/utils.py

```diff
@@ -1700,40 +1700,11 @@ def copytree(
         dirs_exist_ok: bool = False,
         ) -> Path:
     """ Similar to shutil.copytree but with dirs_exist_ok for Python < 3.8 """
-    # No need to reimplement for newer python or if argument is not requested
-    # ignore[call-arg] needed when running type-checks with Python < 3.8
-    if not dirs_exist_ok or sys.version_info >= (3, 8):
-        return cast(
-            Path,
-            shutil.copytree(src=src, dst=dst, symlinks=symlinks,  # type: ignore[call-arg]
-                            dirs_exist_ok=dirs_exist_ok))
-    # Choice was to either copy python implementation and change ONE line
-    # or use rsync (or cp with shell)
-    # We need to copy CONTENT of src into dst
-    # so src has to end with / and dst cannot
-    rsync_src, rsync_dst = str(src), str(dst)
-    if rsync_src[-1] != '/':
-        rsync_src += '/'
-    if rsync_dst[-1] == '/':
-        rsync_dst = rsync_dst[:-1]
-
-    dst.mkdir(parents=True, exist_ok=dirs_exist_ok)
-    command = ["rsync", "-r"]
-    if symlinks:
-        command.append('-l')
-    command.extend([rsync_src, rsync_dst])
-
-    log.debug(f"Calling command '{command}'.")
-    outcome = subprocess.run(
-        command,
-        stdin=subprocess.DEVNULL, stdout=subprocess.PIPE,
-        stderr=subprocess.STDOUT, universal_newlines=True)
-
-    if outcome.returncode != 0:
-        raise shutil.Error(
-            [f"Unable to copy '{src}' into '{dst}' using rsync.",
-             outcome.returncode, outcome.stdout])
-    return dst
+    # FIXME fix all usages, we don't need this function any more
+    return cast(
+        Path,
+        shutil.copytree(src=src, dst=dst, symlinks=symlinks,
+                        dirs_exist_ok=dirs_exist_ok))
```

Comment:
Note that https://github.com/teemtee/tmt/pull/2209/commits/0f6ecbd6ec77bcde831046a612e6bfaf5d69d81 icked more dramatic approach and replaced calls of `tmt.utils.copytree` with `shutil.copytree`. It seemed better to me after this function turned into a very thin wrapper.

Reply:
yes. However I don't want to break tmt API in this PR.
 With all the changes of 2209 api changes are somehow more acceptable IMO.

Reply:
Fair enough. I can see that change as a dedicated PR, easily.